Made entity listener documentation less ambiguous

romaricdrigon opened the pull request:
Hello,

Entity listener documentation was giving different syntaxes, without explaining was was required exactly and for which versions to use those. It was hard to understand the entity annotation is required if you use the "short" service definition, or that you can use the "long" comprehensive service definition in place.
It was misleading for new users, and even for senior ones, I got it wrong today.

This change proposal hopes to make it clearer, but explaining the 2 different syntaxes one after the other.

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -1,10 +1,28 @@
-be added to the resolver. Use the entity_manager attribute to specify which
-entity manager it should be registered with. Example:
+Entity listeners that are services must be registered with the entity listener resolver. 
+On top of the annotation in the entity class, you have to tag the service with "doctrine.orm.entity_listener" for it to be automatically be added to the resolver.
```
```suggestion
On top of the annotation in the entity class, you have to tag the service with
`doctrine.orm.entity_listener` for it to be automatically be added to the resolver.
```

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -1,10 +1,28 @@
-entity manager it should be registered with. Example:
+Entity listeners that are services must be registered with the entity listener resolver. 
+On top of the annotation in the entity class, you have to tag the service with "doctrine.orm.entity_listener" for it to be automatically be added to the resolver.
+Use the (optional) entity_manager attribute to specify which entity manager it should be registered with. 
```
```suggestion
Use the (optional) `entity_manager` attribute to specify which entity manager it should be registered with. 
```

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -14,11 +32,43 @@ entity manager it should be registered with. Example:
+            </services>
+        </container>
+        
+Starting with doctrine/orm 2.5 and Doctrine bundle 1.5.2, instead of registering the entity listener on the entity, you can declare all options from the service definition:
```
```suggestion
Starting with doctrine/orm 2.5 and Doctrine bundle 1.5.2, instead of
registering the entity listener on the entity, you can declare all
options from the service definition:
```

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -14,11 +32,43 @@ entity manager it should be registered with. Example:
                 tags:
+                    # Minimal configuration
                     - { name: doctrine.orm.entity_listener }
+                    # Or, optionnally, you can give the entity manager name
```
```suggestion
                    # Or, optionally, you can give the entity manager name
```

romaricdrigon commented:
Thank you for the proofreading @greg0ire 
I will squash all commits together.

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -1,10 +1,29 @@
-entity manager it should be registered with. Example:
+Entity listeners that are services must be registered with the entity listener resolver. 
+On top of the annotation in the entity class, you have to tag the service with
+`doctrine.orm.entity_listener` for it to be automatically added to the resolver.
```
This is rst, sorry, I forgot extra backticks
```suggestion
``doctrine.orm.entity_listener`` for it to be automatically added to the resolver.
```

greg0ire commented on Resources/doc/entity-listeners.rst:
```diff
@@ -1,10 +1,29 @@
+Entity listeners that are services must be registered with the entity listener resolver. 
+On top of the annotation in the entity class, you have to tag the service with
+`doctrine.orm.entity_listener` for it to be automatically added to the resolver.
+Use the (optional) `entity_manager` attribute to specify which entity manager it should be registered with. 
```
```suggestion
Use the (optional) ``entity_manager`` attribute to specify which entity manager it should be registered with. 
```

romaricdrigon commented:
Sure @alcaeus , here you are :)

alcaeus commented on Resources/doc/entity-listeners.rst:
```diff
@@ -14,11 +33,45 @@ entity manager it should be registered with. Example:
+
+            <services>
+                <service id="user_listener" class="UserListener">
+                    <tag name="doctrine.orm.entity_listener" event="preUpdate" entity="App\Entity\User" />
```
Since we're using the minimal configuration in YAML, the same configuration should be given here as well.

romaricdrigon replied:
Fixed

romaricdrigon commented:
Hello @alcaeus,
I rebased and updated my PR. Reading again #906, I realized having 2 configuration example (one without entity manager option, one with) could be misleading. So I simplified it using comments, I hope it makes more sense.

romaricdrigon commented:
The symfony.com documentation (https://symfony.com/doc/master/bundles/DoctrineBundle/entity-listeners.html) was never updated, do you know why? Is it based on another branch (master?)?

alcaeus commented:
I can take a look at that next week, will be out for a few days for personal reasons.

romaricdrigon commented:
Sure, thank you

romaricdrigon commented:
I saw the Symfony website was updated, thanks :)
